<?xml version="1.0"?>
<launch>

	<arg name="show_images_from_cameras" default="true" />

	<arg name="video_device_right" default="/dev/v4l/by-path/pci-0000:00:14.0-usb-0:5:1.0-video-index0" />
	<arg name="video_device_left" default="/dev/v4l/by-path/pci-0000:00:14.0-usb-0:6:1.0-video-index0" />
	<!-- <arg name="video_device_right" default="/dev/video2" /> -->
	<!-- <arg name="video_device_left" default="/dev/video0" />  -->

	<arg name="image_width" default="640" />
	<arg name="image_height" default="480" />

	<arg name="right_image_namespace" default="stereo" />
	<arg name="left_image_namespace" default="stereo" />

	<arg name="camera_name" default="stereo" />	
	<arg name="publish_tf_period" default="1000" />	

	<arg name="right_image_node" default="right" />
	<arg name="left_image_node" default="left" />

	<arg name="image_topic_name_right" default="image_raw" />
	<arg name="camera_info_topic_name_right" default="camera_info" />

	<arg name="image_topic_name_left" default="image_raw" />
	<arg name="camera_info_topic_name_left" default="camera_info" />

	<arg name="left_camera_name" default="left" />
	<arg name="right_camera_name" default="right" />

	<arg name="left_camera_frame_id" default="left_camera" />
	<arg name="right_camera_frame_id" default="right_camera" />

	<arg name="framerate" default="30" />
	<arg name="pixel_format" default="yuyv" />

	<arg name="camera_info" default="true" />
	<arg name="camera_info_path" default="xiaomi_wooden_stereo" />

	<arg name="stereo_params_path" default="$(find tigra_vision)/config/head_stereo.yaml" />

	<node ns="$(arg camera_name)" name="$(arg right_image_node)" pkg="usb_cam" type="usb_cam_node" output="screen" >
		<param name="video_device" value="$(arg video_device_right)" />
		<param name="image_width" value="$(arg image_width)" />
		<param name="image_height" value="$(arg image_height)"/>
		<param name="pixel_format" value="$(arg pixel_format)" />
		<param name="io_method" value="mmap"/>
		<!-- <remap from="/usb_cam/image_raw" to="$(arg image_topic_name_right)" /> -->
		<param name="framerate" value="$(arg framerate)"/> 
		<!-- if camera_info is available, we will use it-->
		<param name="camera_frame_id" value="$(arg camera_name)_right_optical_frame" if="$(arg camera_info)" /> 
		<param name="camera_info_url" value="file://$(find tigra_vision)/calib_info/$(arg camera_info_path)/right.yaml" if="$(arg camera_info)"/>
		<param name="camera_name" value="$(arg right_camera_name)" if="$(arg camera_info)"/> 
		<!-- <remap from="/usb_cam/camera_info" to="$(arg camera_info_topic_name_right)" if="$(arg camera_info)"/> -->

	</node>

	<node ns="$(arg camera_name)" name="$(arg left_image_node)" pkg="usb_cam" type="usb_cam_node" output="screen" >
		<param name="video_device" value="$(arg video_device_left)" />
		<param name="image_width" value="$(arg image_width)" />
		<param name="image_height" value="$(arg image_height)"/>
		<param name="pixel_format" value="$(arg pixel_format)" />
		<param name="io_method" value="mmap"/>
		<param name="framerate" value="$(arg framerate)"/> 
		<!-- <remap from="/usb_cam/image_raw" to="$(arg image_topic_name_left)"/> -->
		<!-- if camera_info is available, we will use it-->
		<param name="camera_frame_id" value="$(arg camera_name)_left_optical_frame" if="$(arg camera_info)"/> 
		<param name="camera_name" value="$(arg left_camera_name)" if="$(arg camera_info)"/> 
		<param name="camera_info_url" value="file://$(find tigra_vision)/calib_info/$(arg camera_info_path)/left.yaml" if="$(arg camera_info)"/> 
		<!-- <remap from="/usb_cam/camera_info" to="$(arg camera_info_topic_name_left)" if="$(arg camera_info)"/> -->
	</node>

	<node pkg="tf" type="static_transform_publisher" name="tf_$(arg camera_name)_link_2_$(arg camera_name)_left" args="0 0.1 0 0 0 0 $(arg camera_name)_link $(arg camera_name)_left $(arg publish_tf_period)" />
	<node pkg="tf" type="static_transform_publisher" name="tf_$(arg camera_name)_link_2_$(arg camera_name)_right" args="0 -0.1 0 0 0 0 $(arg camera_name)_link $(arg camera_name)_right $(arg publish_tf_period)" />

	<node pkg="tf" type="static_transform_publisher" name="tf_$(arg camera_name)_link_2_$(arg camera_name)_left_optical" args="0 0 0 -1.570796 0 -1.570796 $(arg camera_name)_left $(arg camera_name)_left_optical_frame $(arg publish_tf_period)" />
	<node pkg="tf" type="static_transform_publisher" name="tf_$(arg camera_name)_link_2_$(arg camera_name)_right_optical" args="0 0 0 -1.570796 0 -1.570796 $(arg camera_name)_right $(arg camera_name)_right_optical_frame $(arg publish_tf_period)" />

	<node if="$(arg camera_info)" ns="$(arg camera_name)" pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen">
		<rosparam file="$(arg stereo_params_path)" command="load" />
      	<param name="approximate_sync" value="true" />
	</node>

	<group if="$(arg show_images_from_cameras)">

	<include file="$(find tigra_vision)/launch/view/view_left_right_image.launch" />
	</group>
</launch>
